core/module/name_spec.rb

```diff
@@ -32,6 +32,19 @@ module m::N; end
     m::N.name.should == "ModuleSpecs::Anonymous::WasAnnon"
   end
 
+  it "may be the repeated in different module objects" do
+    m = Module.new
+    n = Module.new
+
+    verbose, $VERBOSE = $VERBOSE, nil
+    ModuleSpecs::Anonymous::SameName = m
+    ModuleSpecs::Anonymous::SameName = n
+    $VERBOSE = verbose
```

Comment:
Could you use `suppress_warning do` here?

Reply:
Ahhh, looked for a helper like this. All updated!